pkg/function-manager/handlers.go

```diff
@@ -456,6 +454,9 @@ func (h *Handlers) updateFunction(params fnstore.UpdateFunctionParams, principal
 		})
 	}
 
+	// generating a new UUID will force the creation of a new function in the underlying FaaS
+	id := uuid.NewV4().String()
+	e.BaseEntity.ID = id
```

Comment:
I thought I already commented on this, but we shouldn't change the ID of an entity (they should be considered immutable).  What we should do is add another property like "FaaSID" which is used as the name for the function within the FaaS.

Reply:
You definitely did make this comment. This is what I have been working to fix. I'm currently stuck trying to figure out why the deployment in OpenFaaS is never ready even though the pod seems to indicate that it is 'serving'. I will ping Ivan.